src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -558,6 +563,10 @@ internal HttpRequestMessage BuildSetTriggersRequest()
 
             var hostname = _hostNameProvider.Value;
             var url = $"{protocol}://{hostname}/operations/settriggers";
+            if (_environment.IsKubernetesManagedHosting())
```

Comment:
nit: can you make this an if/else, so we don't form the url above only to overwrite it?